.github/workflows/remote-copy.yml

```diff
@@ -69,12 +69,8 @@ jobs:
         run: |
           errors=false
 
-          if [ -z "${{ inputs.source }}"]; then
-            echo "::error::No 'source' input given, can't copy anything."
-            errors=true
-          fi
-          if [ -z "${{ inputs.target }}"]; then
-            echo "::error::No 'target' input given, can't copy to anywhere."
+          if [[ "$(basename "${{ inputs.source }}")" == "$(basename "${{ inputs.target }}")" ]]; then
+            echo "::error::Attempted to copy a file into a file (destination must be a directory) or copy a directory into one with the same name (common rsync error)"
```

Comment:
This has the side-effect of not allowing a file with **exactly the same name** as its enclosing directory. 

This is no bad thing IMO.